diff --git a/src/components/mainpage/MainNavbar.js b/src/components/mainpage/MainNavbar.js
index bdc45b1..e8ca030 100644
--- a/src/components/mainpage/MainNavbar.js
+++ b/src/components/mainpage/MainNavbar.js
@@ -7,7 +7,6 @@ import Search from 'components/mainpage/Search';
 const MainNavbar = () => (
   <React.Fragment>
     <Toolbar
-      fixed
       inset
       nav={<MenuButtonContainer className="MenuButton" />}
       title={<Search />}
diff --git a/src/components/mainpage/Mainpage.js b/src/components/mainpage/Mainpage.js
index 578c07c..18290c7 100644
--- a/src/components/mainpage/Mainpage.js
+++ b/src/components/mainpage/Mainpage.js
@@ -22,14 +22,16 @@ const accessibilityProps = {
 
 class Mainpage extends React.Component {
   render() {
-    if (this.props.ownuserIsLoading) {
+    // categories related state is here because we need to render
+    // the categories which located on Home, which is basically our first views
+    if (this.props.ownuserIsLoading || this.props.categoriesIsLoading) {
       // for accessibility
       accessibilityProps['aria-busy'] = false;
       return (
         <CircularProgress id={accessibilityProps['aria-describedby']} />
       );
     }
-    if (this.props.ownuserHasErrored) {
+    if (this.props.ownuserHasErrored || this.props.categoriesHasErrored) {
       return (
         <h2>Error</h2>
       );
@@ -57,8 +59,8 @@ Mainpage.propTypes = {
   ownuserIsLoading: PropTypes.bool.isRequired,
   ownuserHasErrored: PropTypes.bool.isRequired,
   isLogin: PropTypes.bool.isRequired,
-  // categoriesIsLoading: PropTypes.bool.isRequired,
-  // categoriesHasErrored: PropTypes.bool.isRequired,
+  categoriesIsLoading: PropTypes.bool.isRequired,
+  categoriesHasErrored: PropTypes.bool.isRequired,
 };
 
 export default Mainpage;
diff --git a/src/components/views/categories/Categories.js b/src/components/views/categories/Categories.js
index e4f58ff..655bd81 100644
--- a/src/components/views/categories/Categories.js
+++ b/src/components/views/categories/Categories.js
@@ -1,33 +1,33 @@
 import React from 'react';
 import { withRouter } from 'react-router-dom';
 import PropTypes from 'prop-types';
-import { CircularProgress } from 'react-md';
-import urls from 'constants/urls';
+// import { CircularProgress } from 'react-md';
+// import urls from 'constants/urls';
 import CategoryContent from 'components/views/categories/CategoryContent';
 import BackButton from 'components/views/BackButton';
 import ViewsContainer from 'containers/views/ViewsContainer';
 
-const accessibilityProps = {
-  'aria-busy': true,
-  'aria-describedby': 'categories-loading-progress',
-};
+// const accessibilityProps = {
+//   'aria-busy': true,
+//   'aria-describedby': 'categories-loading-progress',
+// };
 
 class Categories extends React.Component {
-  componentDidMount() {
-    this.props.fetchData(urls.categories.url);
-  }
+  // componentDidMount() {
+  //   this.props.fetchData(urls.categories.url);
+  // }
 
   render() {
-    let content = null;
+    const content = <CategoryContent categories={this.props.categories} />;
 
-    if (this.props.categoriesHasErrored) {
-      content = <h1>Categories Display Error!</h1>;
-    } else if (this.props.categoriesIsLoading) {
-      accessibilityProps['aria-busy'] = false;
-      content = <CircularProgress id={accessibilityProps['aria-describedby']} />;
-    } else {
-      content = <CategoryContent categories={this.props.categories} />;
-    }
+    // if (this.props.categoriesHasErrored) {
+    // content = <h1>Categories Display Error!</h1>;
+    // } else if (this.props.categoriesIsLoading) {
+    // accessibilityProps['aria-busy'] = false;
+    // content = <CircularProgress id={accessibilityProps['aria-describedby']} />;
+    // } else {
+    // content = <CategoryContent categories={this.props.categories} />;
+    // }
 
     return (
       <ViewsContainer
@@ -43,11 +43,11 @@ class Categories extends React.Component {
 }
 
 Categories.propTypes = {
-  fetchData: PropTypes.func.isRequired,
+  // fetchData: PropTypes.func.isRequired,
   categories: PropTypes.arrayOf(PropTypes.object).isRequired,
   location: PropTypes.object.isRequired,
-  categoriesHasErrored: PropTypes.bool.isRequired,
-  categoriesIsLoading: PropTypes.bool.isRequired,
+  // categoriesHasErrored: PropTypes.bool.isRequired,
+  // categoriesIsLoading: PropTypes.bool.isRequired,
 };
 
 export default withRouter(Categories);
diff --git a/src/components/views/home/CategorySelector.js b/src/components/views/home/CategorySelector.js
index c47a79f..c29e0e0 100644
--- a/src/components/views/home/CategorySelector.js
+++ b/src/components/views/home/CategorySelector.js
@@ -1,11 +1,19 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import { Card, Avatar, Button } from 'react-md';
+import { Card, Button } from 'react-md';
 
 class CategorySelector extends Component {
   render() {
     const content = this.props.categories.map(category => (
-      <Avatar icon={<Button flat icon primary>{category.mdIcon}</Button>} />
+      <Button
+        icon
+        primary
+        forceIconSize="50px"
+        forceIconFontSize
+        className="CategorySelectorItem"
+      >
+        {category.mdIcon}
+      </Button>
     ));
 
     return (
diff --git a/src/components/views/home/Home.js b/src/components/views/home/Home.js
index b9b987c..0228f97 100644
--- a/src/components/views/home/Home.js
+++ b/src/components/views/home/Home.js
@@ -1,12 +1,21 @@
 import React from 'react';
+import { Drawer } from 'react-md';
 import MainNavbar from 'components/mainpage/MainNavbar';
 import SidebarContainer from 'containers/sidebar/SidebarContainer';
 import CategorySelectorContainer from 'containers/views/home/CategorySelectorContainer';
+import HomeContent from 'components/views/home/HomeContent';
 
 const Home = () => (
   <React.Fragment>
-    <MainNavbar />
-    <CategorySelectorContainer />
+    <Drawer
+      type={Drawer.DrawerTypes.TEMPORARY}
+      defaultVisible
+      position="right"
+    >
+      <MainNavbar />
+      <CategorySelectorContainer />
+      <HomeContent />
+    </Drawer>
     <SidebarContainer />
   </React.Fragment>
 );
diff --git a/src/components/views/home/HomeContent.js b/src/components/views/home/HomeContent.js
new file mode 100644
index 0000000..811d358
--- /dev/null
+++ b/src/components/views/home/HomeContent.js
@@ -0,0 +1,9 @@
+import React from 'react';
+import { Grid } from 'react-md';
+
+const HomeContent = () => (
+  <Grid className="HomeContent">
+  </Grid>
+);
+
+export default HomeContent;
diff --git a/src/components/views/login/Login.js b/src/components/views/login/Login.js
index 86845a0..2bbb400 100644
--- a/src/components/views/login/Login.js
+++ b/src/components/views/login/Login.js
@@ -11,6 +11,7 @@ class Login extends React.Component {
 
   handleClick() {
     this.props.fetchData(urls.ownUser.url);
+    this.props.categoriesfetchData(urls.categories.url);
   }
 
   render() {
@@ -32,6 +33,7 @@ class Login extends React.Component {
 
 Login.propTypes = {
   fetchData: PropTypes.func.isRequired,
+  categoriesfetchData: PropTypes.func.isRequired,
 };
 
 export default Login;
diff --git a/src/constants/tempHomeData.json b/src/constants/tempHomeData.json
new file mode 100644
index 0000000..24df4c0
--- /dev/null
+++ b/src/constants/tempHomeData.json
@@ -0,0 +1,235 @@
+[
+  {
+    "id": "1",
+    "createdAt": 1523148996,
+    "name": "Baby Stroller",
+    "imageUrl": [
+      "https://s25.postimg.org/yrur7v4dr/home-item-baby-1.jpg"
+    ],
+    "description": "Great baby stroller",
+    "price": 100000,
+    "type": "baby",
+    "username": "Flies",
+    "avatarUrl": "https://s25.postimg.org/nfd25ei0f/bug-avatar-1.jpg"
+  },
+  {
+    "id": "2",
+    "createdAt": 1523148936,
+    "name": "Pile of books",
+    "imageUrl": ["https://s25.postimg.org/z4m5e1exr/home-item-book-1.jpg"],
+    "description": "An antique books!",
+    "price": 5000000,
+    "type": "book",
+    "username": "Cathy",
+    "avatarUrl": "https://s25.postimg.org/kdcmkgyun/cat-avatar.jpg"
+  },
+  {
+    "id": "3",
+    "createdAt": 1523148876,
+    "name": "Nice Book",
+    "imageUrl": ["https://s25.postimg.org/6rqnnl8n3/home-item-book-2.jpg"],
+    "description": "description 3",
+    "price": "price 3",
+    "type": "type 3",
+    "username": "username 3",
+    "avatarUrl": "avatarUrl 3"
+  },
+  {
+    "id": "4",
+    "createdAt": 1523148816,
+    "name": "Vintage Car",
+    "imageUrl": ["https://s25.postimg.org/itm1hpuq7/home-item-car-1.jpg"],
+    "description": "description 4",
+    "price": "price 4",
+    "type": "type 4",
+    "username": "username 4",
+    "avatarUrl": "avatarUrl 4"
+  },
+  {
+    "id": "5",
+    "createdAt": 1523148756,
+    "name": "Smart Phone",
+    "imageUrl": ["https://s25.postimg.org/k8nm6gqof/home-item-electronics-1.jpg", "https://s25.postimg.org/74i1tslrz/home-item-electronics-2.jpg", "https://s25.postimg.org/fmrhy4cv3/home-item-electronics-3.jpg"],
+    "description": "description 5",
+    "price": "price 5",
+    "type": "type 5",
+    "username": "username 5",
+    "avatarUrl": "avatarUrl 5"
+  },
+  {
+    "id": "6",
+    "createdAt": 1523148696,
+    "name": "Rolling Phone",
+    "imageUrl": ["https://s25.postimg.org/6rqnnlvsf/home-item-electronics-4.jpg"],
+    "description": "description 6",
+    "price": "price 6",
+    "type": "type 6",
+    "username": "username 6",
+    "avatarUrl": "avatarUrl 6"
+  },
+  {
+    "id": "7",
+    "createdAt": 1523148636,
+    "name": "Knitting Package",
+    "imageUrl": ["https://s25.postimg.org/i4395f727/home-item-fashion-1.jpg", "https://s25.postimg.org/4n6amjhb3/home-item-fashion-2.jpg"],
+    "description": "description 7",
+    "price": "price 7",
+    "type": "type 7",
+    "username": "username 7",
+    "avatarUrl": "avatarUrl 7"
+  },
+  {
+    "id": "8",
+    "createdAt": 1523148576,
+    "name": "Blue Blazer",
+    "imageUrl": ["https://s25.postimg.org/abcldfdxr/home-item-fashion-3.jpg"],
+    "description": "description 8",
+    "price": "price 8",
+    "type": "type 8",
+    "username": "username 8",
+    "avatarUrl": "avatarUrl 8"
+  },
+  {
+    "id": "9",
+    "createdAt": 1523148516,
+    "name": "Pack of Marbles",
+    "imageUrl": ["https://s25.postimg.org/ex8plt21r/home-item-free-1.jpg"],
+    "description": "description 9",
+    "price": "price 9",
+    "type": "type 9",
+    "username": "username 9",
+    "avatarUrl": "avatarUrl 9"
+  },
+  {
+    "id": "10",
+    "createdAt": 1523148456,
+    "name": "Dangerous Drums",
+    "imageUrl": ["https://s25.postimg.org/t3ogh1573/home-item-free-2.jpg"],
+    "description": "description 10",
+    "price": "price 10",
+    "type": "type 10",
+    "username": "username 10",
+    "avatarUrl": "avatarUrl 10"
+  },
+  {
+    "id": "11",
+    "createdAt": 1523148396,
+    "name": "Deserted Car",
+    "imageUrl": ["https://s25.postimg.org/jvw80bipb/home-item-free-3.jpg"],
+    "description": "description 11",
+    "price": "price 11",
+    "type": "type 11",
+    "username": "username 11",
+    "avatarUrl": "avatarUrl 11"
+  },
+  {
+    "id": "12",
+    "createdAt": 1523148336,
+    "name": "Fresh Orange",
+    "imageUrl": ["https://s25.postimg.org/gcaaakl5b/home-item-home-1.jpg"],
+    "description": "description 12",
+    "price": "price 12",
+    "type": "type 12",
+    "username": "username 12",
+    "avatarUrl": "avatarUrl 12"
+  },
+  {
+    "id": "13",
+    "createdAt": 1523148276,
+    "name": "Red Apple",
+    "imageUrl": ["https://s25.postimg.org/h1t2mxten/home-item-home-2.jpg"],
+    "description": "description 13",
+    "price": "price 13",
+    "type": "type 13",
+    "username": "username 13",
+    "avatarUrl": "avatarUrl 13"
+  },
+  {
+    "id": "14",
+    "createdAt": 1523148216,
+    "name": "name 14",
+    "imageUrl": [],
+    "description": "description 14",
+    "price": "price 14",
+    "type": "type 14",
+    "username": "username 14",
+    "avatarUrl": "avatarUrl 14"
+  },
+  {
+    "id": "15",
+    "createdAt": 1523148156,
+    "name": "name 15",
+    "imageUrl": [],
+    "description": "description 15",
+    "price": "price 15",
+    "type": "type 15",
+    "username": "username 15",
+    "avatarUrl": "avatarUrl 15"
+  },
+  {
+    "id": "16",
+    "createdAt": 1523148096,
+    "name": "name 16",
+    "imageUrl": [],
+    "description": "description 16",
+    "price": "price 16",
+    "type": "type 16",
+    "username": "username 16",
+    "avatarUrl": "avatarUrl 16"
+  },
+  {
+    "id": "17",
+    "createdAt": 1523148036,
+    "name": "name 17",
+    "imageUrl": [],
+    "description": "description 17",
+    "price": "price 17",
+    "type": "type 17",
+    "username": "username 17",
+    "avatarUrl": "avatarUrl 17"
+  },
+  {
+    "id": "18",
+    "createdAt": 1523147976,
+    "name": "name 18",
+    "imageUrl": [],
+    "description": "description 18",
+    "price": "price 18",
+    "type": "type 18",
+    "username": "username 18",
+    "avatarUrl": "avatarUrl 18"
+  },
+  {
+    "id": "19",
+    "createdAt": 1523147916,
+    "name": "name 19",
+    "imageUrl": [],
+    "description": "description 19",
+    "price": "price 19",
+    "type": "type 19",
+    "username": "username 19",
+    "avatarUrl": "avatarUrl 19"
+  },
+  {
+    "id": "20",
+    "createdAt": 1523147856,
+    "name": "name 20",
+    "imageUrl": [],
+    "description": "description 20",
+    "price": "price 20",
+    "type": "type 20",
+    "username": "username 20",
+    "avatarUrl": "avatarUrl 20"
+  },
+  {
+    "id": "21",
+    "createdAt": 1523147796,
+    "name": "name 21",
+    "imageUrl": [],
+    "description": "description 21",
+    "price": "price 21",
+    "type": "type 21",
+    "username": "username 21",
+    "avatarUrl": "avatarUrl 21"
+  }
+]
\ No newline at end of file
diff --git a/src/containers/views/login/LoginContainer.js b/src/containers/views/login/LoginContainer.js
index 28078c9..d4961bf 100644
--- a/src/containers/views/login/LoginContainer.js
+++ b/src/containers/views/login/LoginContainer.js
@@ -1,6 +1,7 @@
 import { connect } from 'react-redux';
 import Login from 'components/views/login/Login';
 import { ownuserFetchData } from 'actions/views/loginActions';
+import { categoriesFetchData } from 'actions/views/categoriesActions';
 
 // const mapStateToProps = state => ({
 //   ownuser: state.ownuser,
@@ -12,6 +13,9 @@ const mapDispatchToProps = dispatch => ({
   fetchData: (url) => {
     dispatch(ownuserFetchData(url));
   },
+  categoriesfetchData: (url) => {
+    dispatch(categoriesFetchData(url));
+  },
 });
 
 const LoginContainer = connect(
diff --git a/src/styles/components/views/home/_CategorySelector.scss b/src/styles/components/views/home/_CategorySelector.scss
index 5f08809..55251de 100644
--- a/src/styles/components/views/home/_CategorySelector.scss
+++ b/src/styles/components/views/home/_CategorySelector.scss
@@ -1,4 +1,12 @@
 .CategorySelector {
   display: flex;
+  align-items: center;
   overflow-x: scroll;
+  height: 8rem;
+}
+
+button.CategorySelectorItem i {
+  height: 5rem;
+  width: 5rem;
+  font-size: 4rem;
 }
\ No newline at end of file
